plan/selectivity.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"math"
+
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/model"
+)
+
+// indexBlock is used when calculating selectivity
+type indexBlock struct {
+	// the position that this index is in the index slice, if is -1, then this is primary key.
+	pos int
+	// the ith bit of `cover` will tell whether the ith expression is covered by this index.
+	cover uint64
+	// pk won't use this, only index will use.
```

Comment:
Could you also describe the usage of `inAndEqCnt` like previous two? Because the meaning of it is not obvious in this context.